src/main/scala/Battle.scala

```diff
@@ -0,0 +1,136 @@
+import scala.util.Random
+import scala.io.StdIn.{readLine,readInt}
+
+class Battle{
+  val r: Random.type = scala.util.Random
+  def action(){
+    println(s"Player health:${Player.health}\nPlayer attacks strength:${Player.attacks}\nEnemy health:${Enemy.health}\nEnemy strength:${Enemy.attacks}")
+    println("\nAction menu\n1. Attack\n2. Defend\n3. Spells \n4.Run away")
+    val choice = readInt
+    choice match {
+      case 1 => this.attack()
+      case 2 => this.defend()
+      case 3 => this.spells()
+      case 4 => println("TACTICAL RETREAT")
+      case _ => this.action()
+    }
+  }
+  def attack(){
```

Comment:
this is a good example of a bad method, it's way too big and complicated

you should split it up into helper methods to reduce complexity, now it's not possible to understand what the hell is happening here really